data/mods/ssb/moves.js

```diff
@@ -296,6 +296,133 @@ let BattleMovedex = {
 		secondary: null,
 		target: "self",
 		type: "Psychic",
+	},	
+	// Andrew
+	backoffgrrr: {
+		accuracy: true,
+		basePower: 0,
+		category: "Status",
+		desc: "The user is protected from most moves made by other Pokemon during this turn, and if attacked, the Pokemon is forced to switch to a random ally. This move has a 1/X chance of being successful, where X starts at 1 and doubles each time this move is successfully used. X resets to 1 if this move fails, if the user's last move used is not Baneful Bunker, Detect, Endure, King's Shield, Protect, Quick Guard, Spiky Shield, Wide Guard, or this move, or if it was one of those moves and the user's protection was broken. Fails if the user moves last this turn.",
+		shortDesc: "Protects from moves. Attacked: Force switch foe.",
+		id: "backoffgrrr",
+		name: "Back Off! GRRR!",
+		isNonstandard: "Custom",
+		pp: 15,
+		priority: 4,
+		flags: {},
+		onTryMove() {
+			this.attrLastMove('[still]');
+		},
+		onPrepareHit(target, source) {
+			this.add('-anim', source, "Protect", source);
+			this.add('-anim', source, "Defense Curl", source);
+		},
+		stallingMove: true,
+		volatileStatus: 'backoffgrrr',
+		onTryHit(target, source, move) {
+			return !!this.willAct() && this.runEvent('StallMove', target);
+		},
+		onHit(pokemon) {
+			pokemon.addVolatile('stall');
+		},
+		effect: {
+			duration: 1,
+			onStart(target) {
+				this.add('-singleturn', target, 'move: Protect');
+			},
+			onTryHitPriority: 3,
+			onTryHit(target, source, move) {
+				if (!move.flags['protect']) {
+					if (move.isZ) target.getMoveHitData(move).zBrokeProtect = true;
+					return;
+				}
+				this.add('-activate', target, 'move: Protect');
+				let lockedmove = source.getVolatile('lockedmove');
+				if (lockedmove) {
+					// Outrage counter is reset
+					if (source.volatiles['lockedmove'].duration === 2) {
+						delete source.volatiles['lockedmove'];
+					}
+				}
+				this.add('-anim', target, "Scary Face", source);
+				this.add('-anim', target, "Roar", source);
+				source.forceSwitchFlag = true;
+				return null;
+			},
+			onHit(target, source, move) {
+				if (move.isZPowered) {
```

Comment:
Move should force switch even if it failed to bypass protect from what im reading in the thread.

Reply:
also,  check line 348